overlord/servicestate/quota_control.go

```diff
@@ -98,6 +102,10 @@ func CreateQuota(st *state.State, name string, parentName string, snaps []string
 	if err := resourceLimits.Validate(); err != nil {
 		return nil, fmt.Errorf("cannot create quota group %q: %v", name, err)
 	}
+	// validate that the system has the features needed for this resource
+	if err := resourcesCheckFeatureRequirements(&resourceLimits); err != nil {
```

Comment:
these looks like the right placea as they are also the ones calling quotaGroupsAvailable